Add backdrop for dropdown

JustinBeBoy opened the pull request:
This PR resolve #581 with detail:
- Add backdrop for dropdown
- Implement pages which using dropdown

JustinBeBoy committed:
custom dropdown backdrop and implement page use

Sirmorrison commented on ui/page/tickets/list_page.go:
```diff
@@ -121,6 +121,9 @@ func (pg *ListPage) Layout(gtx layout.Context) layout.Dimensions {
 					}),
+					layout.Expanded(pg.orderDropDown.BackDrop),
+					layout.Expanded(pg.ticketTypeDropDown.BackDrop),
+					layout.Expanded(pg.walletDropDown.BackDrop),
```
This should be added to the `dropdown` widget as part of the layout method.

Sirmorrison commented on ui/page/transactions_page.go:
```diff
@@ -162,6 +162,9 @@ func (pg *TransactionsPage) Layout(gtx layout.Context) layout.Dimensions {
 			}),
+			layout.Expanded(pg.txTypeDropDown.BackDrop),
+			layout.Expanded(pg.orderDropDown.BackDrop),
+			layout.Expanded(pg.walletDropDown.BackDrop),
```
This should be part of the `dropdown` widget not the page the dropdown is used.

Sirmorrison committed:
add backdrop to tx and ticket page
- add IsOpen() and CloseDropdown() to dropdown widget

Sirmorrison committed:
fix wallet dropdown not opening bug ticket activity page
- add backdrop to ticket activity page
- fix wallet dropdown icon size and alignment with wallet text

beansgum commented on ui/page/tickets/activity_page.go:
```diff
@@ -103,6 +105,13 @@ func (pg *ActivityPage) Layout(gtx layout.Context) layout.Dimensions {
 					}),
+					layout.Expanded(func(gtx C) D {
+						if pg.orderDropDown.IsOpen() || pg.ticketTypeDropDown.IsOpen() || pg.walletDropDown.IsOpen() {
+							return pg.backdrop.Layout(gtx)
```
This implementation requires each page to have a backdrop, the backdrop should be handled by the dropdown instead.

JustinBeBoy committed:
- Create backdrop for dropdown
- Remove backdrops in all pages use dropdown
- Refactor code in dropdown, transaction page, activity page and list page
- Align layout dropdown in pages use it

Sirmorrison commented on ui/page/transactions_page.go:
```diff
@@ -162,38 +162,24 @@ func (pg *TransactionsPage) Layout(gtx layout.Context) layout.Dimensions {
+				return pg.walletDropDown.Layout(gtx, 0)
+			}),
+			layout.Expanded(func(gtx C) D {
+				post := float32(gtx.Constraints.Max.X - 315)
```
rename all occurrence of `post` to `position` or `pos`

Sirmorrison reviewed (changes requested):
![image](https://user-images.githubusercontent.com/27733432/132824743-170c7239-1e66-4077-b698-05effbbfd9e2.png)

 on the ticket page this is what the dropdowns look like at the far right.

Same for ticket activity page.

![image](https://user-images.githubusercontent.com/27733432/132825026-ff2f9542-e71e-4df9-9b9f-9f17807b8f8f.png)

Same for transactions page.
![image](https://user-images.githubusercontent.com/27733432/132824925-82033014-153b-4ce7-bc83-3d07d7ac57ad.png)

Sirmorrison commented on ui/page/tickets/activity_page.go:
```diff
@@ -103,8 +101,16 @@ func (pg *ActivityPage) Layout(gtx layout.Context) layout.Dimensions {
+					}),
+					layout.Expanded(func(gtx C) D {
+						post := float32(gtx.Constraints.Max.X - 315)
+						return pg.ticketTypeDropDown.Layout(gtx, post)
```
So doing troubleshooting, this seems to be where the issue is coming from.

Changing the values as suggested below, give me the screenshots below.
![image](https://user-images.githubusercontent.com/27733432/132838330-2cd80f51-5392-4a09-995b-cd852760a527.png)

```suggestion
					layout.Expanded(func(gtx C) D {
						return pg.walletDropDown.Layout(gtx, 0)
					}),
					layout.Expanded(func(gtx C) D {
						post := float32(gtx.Constraints.Max.X - 450)
						return pg.orderDropDown.Layout(gtx, post)
					}),
					layout.Expanded(func(gtx C) D {
						post := float32(gtx.Constraints.Max.X - 300)
						return pg.ticketTypeDropDown.Layout(gtx, post)
```

I think the issue here has to do with the screen size of the machine. you will need to use `gtx.Px()` for positioning and scaling.

JustinBeBoy committed:
fix issue of different screen resolutions

JustinBeBoy committed:
change layout and resolve different resolutions

JustinBeBoy committed:
caculate and change space between dropdown

Sirmorrison reviewed (approved):
PR looks good.
